src/hotspot/share/opto/graphKit.cpp

```diff
@@ -1892,7 +1894,25 @@ void GraphKit::set_arguments_for_java_call(CallJavaNode* call, bool is_late_inli
       continue;
     } else if (arg->is_InlineType()) {
       // Pass inline type argument via oop to callee
-      arg = arg->as_InlineType()->buffer(this);
+      InlineTypeNode* inline_type = arg->as_InlineType();
+      const ciMethod* caller_method = _method;
+      const ciMethod* method = call->method();
+      ciInstanceKlass* holder = method->holder();
+      const bool is_receiver = i == TypeFunc::Parms;
+      bool must_init_buffer = false;
+      // Do we have a larval receiver for an abstract constructor or the Object constructor (that is not going to be
+      // inlined)?
+      if (is_receiver && inline_type->is_larval() && method->is_object_constructor() &&
+          (holder->is_abstract() || holder->is_java_lang_Object())) {
+        // We normally do not want to initialize the buffer of a larval because we are going to update it anyway which
+        // requires us to create a new buffer. But at this point, we are going to call the super constructor of this
+        // concrete or abstract inline type without inlining it. After that, the larval is completely initialized and
+        // thus not a larval anymore. We therefore need to force an initialization of the buffer to not lose all the
+        // field writes so far in case the buffer needs to be used (e.g. to read from when deoptimizing at runtime) or
+        // further updated in abstract super value class constructors which could have more fields to be initialized.
+        must_init_buffer = true;
+      }
+      arg = inline_type->buffer(this, true, must_init_buffer);
```

Comment:
I think `must_init_buffer` should be true by default and set to false here for larvals, see comments below.